hudi-client/hudi-client-common/src/main/java/org/apache/hudi/io/HoodieWriteHandle.java

```diff
@@ -116,7 +117,8 @@ protected HoodieWriteHandle(HoodieWriteConfig config, String instantTime, String
     this.partitionPath = partitionPath;
     this.fileId = fileId;
     this.writeSchema = AvroSchemaCache.intern(overriddenSchema.orElseGet(() -> getWriteSchema(config)));
-    this.writeSchemaWithMetaFields = AvroSchemaCache.intern(HoodieAvroUtils.addMetadataFields(writeSchema, config.allowOperationMetadataField()));
+    this.hasOperationMetaField = config.allowOperationMetadataField();
+    this.writeSchemaWithMetaFields = AvroSchemaCache.intern(HoodieAvroUtils.addMetadataFields(writeSchema, hasOperationMetaField));
```

Comment:
The flag `hasOperationMetaField` is redundant, the field is already within the `writeSchemaWithMetaFields`.